packages/app/android/src/reactnative/java/io/invertase/firebase/common/SharedUtils.java

```diff
@@ -130,6 +130,31 @@ public static boolean isAppInForeground(Context context) {
     List<ActivityManager.RunningAppProcessInfo> appProcesses = activityManager.getRunningAppProcesses();
     if (appProcesses == null) return false;
 
+    // Check if current activity is a background activity
+    ReactNativeFirebaseJSON json = ReactNativeFirebaseJSON.getSharedInstance();
+    if (json.contains("android_background_activity_names")) {
+      ArrayList<String> backgroundActivities = json.getArrayValue("android_background_activity_names");
+      
+      if (backgroundActivities.size() != 0) {
+        String currentActivity = "";
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+          List<ActivityManager.AppTask> taskInfo = activityManager.getAppTasks();
+          if(taskInfo.size() > 0) {
+            currentActivity = taskInfo.get(0).getTaskInfo().baseActivity.getShortClassName();
+          }
+        } else {
+          List<ActivityManager.RunningTaskInfo> taskInfo = activityManager.getRunningTasks(1);
+          if(taskInfo.size() > 0) {
+            currentActivity = taskInfo.get(0).topActivity.getShortClassName();
+          }
+        }
+
+        if (currentActivity != "" && backgroundActivities.contains(currentActivity)) {
```

Comment:
:thinking: two issues here -

1- using `!=` on java strings asks if they are the exact same in-memory location, not if the strings are equal, you have to use `String.equals()` for string equality. In Javascript even it should be `!==` vs `!=`. 
2- NullPointerExceptions on string comparison are a classic Java issue, so if you use Java `String.equals()` the safe way (to protect against nulls) is to reverse them and use the known-good string first, e.g. `!"".equals(currentActivity)`

```suggestion
        if (!"".equals(currentActivity && backgroundActivities.contains(currentActivity)) {
```

Reply:
Happy with the suggestion but it looks like your missing a `)` at the end of your `equals(currentActivity)` check

Reply:
:facepalm:  yes indeed. The perils of coding in a web text box - good spot - updated - hopefully better?